core/src/utils/framework-delegate.ts

```diff
@@ -86,7 +88,7 @@ export const CoreDelegate = () => {
       BaseComponent.appendChild(el);
 
       await new Promise((resolve) => componentOnReady(el, resolve));
-    } else if (BaseComponent.children.length > 0) {
+    } else if (hasUserDefinedComponent && BaseComponent.children.length > 0) {
```

Comment:
This regresses inline modals in vanilla JS. The `.ion-page` class is no longer added when presenting a modal in http://localhost:3333/src/components/modal/test/inline. (We should probably add a test to check for this)